UPGRADING.md

```diff
@@ -0,0 +1,321 @@
+# Clerk Go SDK upgrade guide
+
+The Clerk Go SDK uses [Semantic Versioning](https://semver.org/)
+for tracking all changes to the library.
+
+Semantic versions take the form of `MAJOR.MINOR.PATCH`. Incrementing
+
+- MAJOR version introduces incompatible API changes
+- MINOR version introduces functionality in a backwards compatible manner
+- PATCH version includes backwards compatible bug fixes
+
+This file contains upgrade notes for all `MAJOR` version bumps.
+
+While there are no plans to sunset older versions of the Go SDK,
+the current `MAJOR` version of `clerk-sdk-go` is actively supported.
+
+Bug fixes and security updates may be added to previous versions, but
+there is no guarantee.
+
+## 1.x.x to 2.x.x
+
+The `v2` version of the Clerk Go SDK is a complete rewrite and introduces
+a lot of breaking changes.
+
+### Minimum Go version
+
+The minimum supported Go version for the `v2` version of the Clerk Go SDK is `1.19`.
+
+### Completely different API
```

Comment:
❓ Not sure exactly what alternative to suggest but just noting in case you can think of better phrasing to describe this section 🤔

Reply:
Perhaps you or others can come up with something better but somethine like `New API and Package Layout` (or `Structure`)?

Reply:
Sounds good to me. Updated. :smile: